dlt/common/libs/sqlglot.py

```diff
@@ -0,0 +1,480 @@
+    DataType.Type.UDECIMAL: "decimal",
+    DataType.Type.UDOUBLE: "decimal",
+    # TEMPORAL
+    DataType.Type.DATE: "date",
+    DataType.Type.DATE32: "date",
+    DataType.Type.DATETIME: "date",
+    DataType.Type.DATETIME2: "date",
+    DataType.Type.DATETIME64: "date",
+    DataType.Type.SMALLDATETIME: "date",
+    DataType.Type.TIMESTAMP: "timestamp",
+    DataType.Type.TIMESTAMPNTZ: "timestamp",
+    DataType.Type.TIMESTAMPLTZ: "timestamp",
+    DataType.Type.TIMESTAMPTZ: "timestamp",
+    DataType.Type.TIMESTAMP_MS: "timestamp",
+    DataType.Type.TIMESTAMP_NS: "timestamp",
+    DataType.Type.TIMESTAMP_S: "timestamp",
+    DataType.Type.TIME: "time",
+    DataType.Type.TIMETZ: "time",
+    # binary
+    DataType.Type.VARBINARY: "binary",
+    # BOOLEAN
+    DataType.Type.BOOLEAN: "bool",
+    # UNKNOWN
+    DataType.Type.UNKNOWN: None,
+}
+
+SQLGLOT_INT_PRECISION = {
+    DataType.Type.TINYINT: 3,
+    DataType.Type.SMALLINT: 5,
+    DataType.Type.MEDIUMINT: 8,
+    DataType.Type.INT: 10,
+    DataType.Type.BIGINT: None,  # 19
+    DataType.Type.INT128: 39,
+    DataType.Type.INT256: 78,
+    DataType.Type.UTINYINT: 3,
+    DataType.Type.USMALLINT: 5,
+    DataType.Type.UMEDIUMINT: 8,
+    DataType.Type.UINT: 10,
+    DataType.Type.UBIGINT: 19,
+    DataType.Type.UINT128: 39,
+    DataType.Type.UINT256: 78,
+}
+
+SQLGLOT_DECIMAL_PRECISION_AND_SCALE = {
+    DataType.Type.BIGDECIMAL: (38, 10),
+    DataType.Type.DECIMAL: (38, 10),
+    DataType.Type.DECIMAL32: (7, 2),
+    DataType.Type.DECIMAL64: (16, 4),
+    DataType.Type.DECIMAL128: (34, 10),
+    DataType.Type.DECIMAL256: (76, 20),
+    DataType.Type.MONEY: (19, 4),
+    DataType.Type.SMALLMONEY: (10, 4),
+    DataType.Type.UDECIMAL: (38, 10),
+}
+
+SQLGLOT_TEMPORAL_PRECISION = {
+    DataType.Type.TIMESTAMP: None,  # default value; default precision varies across DB
+    DataType.Type.TIMESTAMP_S: 0,
+    DataType.Type.TIMESTAMP_MS: 3,
+    DataType.Type.TIMESTAMP_NS: 9,
+}
+
+# NOTE in Snowflake, TIMESTAMPNTZ == DATETIME; is this true for dlt?
+SQLGLOT_HAS_TIMEZONE = {
+    DataType.Type.TIMESTAMP: None,  # default value; False
+    DataType.Type.TIMESTAMPNTZ: False,
+    DataType.Type.TIMESTAMPLTZ: True,
+    DataType.Type.TIMESTAMPTZ: True,
+    DataType.Type.TIMESTAMP_MS: False,
+    DataType.Type.TIMESTAMP_NS: False,
+    DataType.Type.TIMESTAMP_S: False,
+    DataType.Type.TIME: False,  # default value; False
+    DataType.Type.TIMETZ: True,
+}
+
+DLT_TO_SQLGLOT = {
+    "json": DataType.Type.JSON,
+    "text": DataType.Type.TEXT,
+    "double": DataType.Type.DOUBLE,
+    "bool": DataType.Type.BOOLEAN,
+    "date": DataType.Type.DATE,
+    "timestamp": DataType.Type.TIMESTAMP,
+    "bigint": DataType.Type.BIGINT,
+    "binary": DataType.Type.VARBINARY,
+    "time": DataType.Type.TIME,
+    "decimal": DataType.Type.DECIMAL,
+    "wei": DataType.Type.UINT256,
+}
+
+
+# TODO should we raise errors on type conversion or silently convert to correct values?
```

Comment:
Can you give an example for an error? I'm not quite clear when one would appear.